lib/simplecov/exit_codes/minimum_coverage_by_file_check.rb

```diff
@@ -29,9 +31,30 @@ def exit_code
 
       attr_reader :result, :minimum_coverage_by_file
 
-      def covered_percentages
-        @covered_percentages ||=
-          result.covered_percentages.map { |percentage| SimpleCov.round_coverage(percentage) }
+      def coverage_statistics_by_file
+        @coverage_statistics_by_file ||=
+          (res = result.coverage_statistics_by_file).each do |criteria, stats|
+            res[criteria] = stats.map { |stat| SimpleCov.round_coverage(stat.percent) }
+          end
+      end
+
+      def minimum_violations
+        @minimum_violations ||=
+          compute_minimum_violations.select do |achieved|
+            achieved.fetch(:actual) < achieved.fetch(:minimum_expected)
+          end
+      end
+
+      def compute_minimum_violations
```

Comment:
similar these aren't the violations yet so I'd name it something like `compute_minimum_coverage_data` or something as with drop above